src_c/_sdl2/video.pyx

```diff
@@ -0,0 +1,302 @@
+        return SDL_GetWindowID(self._win)
+
+    @property
+    def size(self):
+        """get the size of a window's client area."""
+        cdef int w, h
+        SDL_GetWindowSize(self._win, &w, &h)
+        return (w, h)
+
+    @size.setter
+    def size(self, size):
+        """set the size of a window's client area."""
+        SDL_SetWindowSize(self._win, size[0], size[1])
+
+    @property
+    def position(self):
+        cdef int x, y
+        SDL_GetWindowPosition(self._win, &x, &y)
+        return (x, y)
+
+    @position.setter
+    def position(self, position):
+        """screen coordinates, or WINDOWPOS_CENTERED or WINDOWPOS_UNDEFINED"""
+        cdef int x, y
+        if position == WINDOWPOS_UNDEFINED:
+            x, y = WINDOWPOS_UNDEFINED, WINDOWPOS_UNDEFINED
+        elif position == WINDOWPOS_CENTERED:
+            x, y = WINDOWPOS_CENTERED, WINDOWPOS_CENTERED
+        else:
+            x, y = position
+        SDL_SetWindowPosition(self._win, x, y)
+
+    @property
+    def opacity(self):
+        """get window opacity in (0.0f transparent, 1.0f opaque)"""
+        cdef float opacity
+        if SDL_GetWindowOpacity(self._win, &opacity):
+            raise error()
+        return opacity
+
+    @opacity.setter
+    def opacity(self, opacity):
+        """set window opacity in (0.0f transparent, 1.0f opaque)"""
+        if SDL_SetWindowOpacity(self._win, opacity):
+            raise error()
+
+    def __dealloc__(self):
+        self.destroy()
+
+
+cdef class Texture:
+    def __init__(self, Renderer renderer, surface):
+        """ Create a texture from an existing surface.
+
+        :param renderer Renderer: Rendering context for the texture.
+        :param surface Surface: The surface to create a texture from.
+        """
+        # https://wiki.libsdl.org/SDL_CreateTextureFromSurface
+        if not pgSurface_Check(surface):
+            raise error('2nd argument must be a surface')
+        self.renderer = renderer
+        cdef SDL_Renderer* _renderer = renderer._renderer
+        cdef SDL_Surface *surf_ptr = pgSurface_AsSurface(surface)
+        self._tex = SDL_CreateTextureFromSurface(_renderer,
+                                                 surf_ptr)
+        if not self._tex:
+            raise error()
+        self.width = surface.get_width()
+        self.height = surface.get_height()
+
+    def __dealloc__(self):
+        if self._tex:
+            SDL_DestroyTexture(self._tex)
+
+cdef class Renderer:
+    def __init__(self, Window window, int index=-1,
+                 int accelerated=-1, bint vsync=False,
+                 bint target_texture=False):
+        """ Create a 2D rendering context for a window.
+
+        :param window Window: where rendering is displayed.
+        :param index int: index of rendering driver to initialize,
+                          or -1 to init the first supporting requested options.
+        :param accelerated int: if 1, the renderer uses hardware acceleration.
```

Comment:
I like the accelerated handling of flags! (wasn't sure about it at first).
```python
Renderer(accelerated=0)
Renderer(accelerated=1)
Renderer()
```

Reading those I can tell what accelerated=0 means, and accelerated=1 means. The default of -1 probably won't be specified usually, so I don't think it matters that it is not immediately understandable.

Nice!